sqlglot/parser.py

```diff
@@ -3970,27 +3968,16 @@ def _parse_set_operations(self, this: t.Optional[exp.Expression]) -> t.Optional[
         return this
 
     def _parse_expression(self) -> t.Optional[exp.Expression]:
-        return self._parse_alias(self._parse_conjunction())
+        return self._parse_alias(self._parse_op())
 
-    def _parse_conjunction(self) -> t.Optional[exp.Expression]:
-        this = self._parse_equality()
+    def _parse_op(self) -> t.Optional[exp.Expression]:
+        return self._parse_tokens(self._parse_disjunction, self.ASSIGNMENT)
```

Comment:
Not sure if this is right. Lets chat about this on Slack.

Reply:
did you change this? assignment should be top level precedence

x := a and b 

is equal to

x := (a and b)

Reply:
Ah, I think this is where it gets messy:
`a := c := 2 OR b`
And I think in some dialects the right side can be a statement, like SELECT 1, but the old code didn't support that

Reply:
I don't think we originally intended to support multiple assignments using that syntax. At least it's not something we've tested (IIRC), and seems unimportant for DuckDB (which uses this syntax natively) and SQLMesh.

Reply:
Well I'm going to make it behave like before, just to be safe.